src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1183 @@
+   *
+   * @param currStmt statement to simplify
+   * @param primitiveValues map of primitive variable names to expressions representing their values
+   * @return list of statements, where each is a possible simplification of {@code currStmt}
+   */
+  private static List<Statement> getStatementReplacements(
+      Statement currStmt, Map<String, String> primitiveValues) {
+    List<Statement> replacements = new ArrayList<Statement>();
+
+    // Null represents removal of the statement.
+    replacements.add(null);
+
+    if (currStmt instanceof ExpressionStmt) {
+      Expression exp = ((ExpressionStmt) currStmt).getExpression();
+      if (exp instanceof VariableDeclarationExpr) {
+        // Create and return a list of possible replacement statements.
+        VariableDeclarationExpr vdExpr = (VariableDeclarationExpr) exp;
+
+        // Simplify right hand side to zero-equivalent value: 0, false,
+        // or null.
+        List<Statement> rhsZeroValStmts = rhsAssignZeroValue(vdExpr);
+        replacements.addAll(rhsZeroValStmts);
+
+        // Simplify right hand side to a value that was previously found
+        // in a passing assertion.
+        Statement rhsAssertValStmt = rhsAssignValueFromPassingAssertion(vdExpr, primitiveValues);
+        if (rhsAssertValStmt != null) {
+          replacements.add(rhsAssertValStmt);
+        }
+
+        // Simplify statement by removing the left hand side.
+        Statement lhsRemovalStmt = removeLeftHandSideSimplification(vdExpr);
+        if (lhsRemovalStmt != null) {
+          replacements.add(lhsRemovalStmt);
+        }
+      }
+    }
+    return replacements;
+  }
+
+  /**
+   * Return a list of variable declaration statements that could replace the right hand side by 0,
+   * false, or null, whichever is type correct. Returns an empty list if there are multiple variable
+   * declarations in a single statement, such as {@code int i, j, k;}.
+   *
+   * @param vdExpr variable declaration expression representing the current statement to simplify
+   * @return a list of {@code Statement} objects representing the simplified variable declaration
+   *     expression if the type of the variable is a primitive and a value has been previously
+   *     calculated.
+   */
+  private static List<Statement> rhsAssignZeroValue(VariableDeclarationExpr vdExpr) {
+    List<Statement> resultList = new ArrayList<Statement>();
+
+    if (vdExpr.getVars().size() != 1) {
+      // Number of variables declared in this expression is not 1.
+      return resultList;
+    }
+
+    Type type = vdExpr.getType();
+    if (type instanceof PrimitiveType) {
+      // Replacement with zero value on the right hand side.
+      resultList.add(rhsAssignWithValue(vdExpr, type, null));
+    } else {
+      // Replacement with null on the right hand side.
+      resultList.add(rhsAssignWithValue(vdExpr, type, null));
+
+      if (type instanceof ReferenceType) {
```

Comment:
How can this be false?  If it cannot be, then just do the cast without a check.
If a programmer writes a check, that is a signal to a reader that it could return either true or false at run time.